staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go

```diff
@@ -62,8 +79,27 @@ func TestLoopbackHostPort(t *testing.T) {
 	if ip := net.ParseIP(host); ip == nil || !ip.IsLoopback() || ip.To4() != nil {
 		t.Fatalf("expected IPv6 host to be loopback, got %q", host)
 	}
-
 	if port != "443" {
 		t.Fatalf("expected 443 as port, got %q", port)
 	}
 }
+
+func isIPv6LoopbackSupported() (bool, bool, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err == nil {
+		ipv4, ipv6 := false, false
+		for _, address := range addrs {
+			ipnet, ok := address.(*net.IPNet)
+			if !ok || !ipnet.IP.IsLoopback() {
+				continue
+			}
+			if ipnet.IP.To4() == nil {
```

Comment:
Is it enough to assume IPv6 if `To4()` returns nil? Might be worthwhile to also check that `len(ipnet.IP) == net.IPv6len). Thoughts?

Reply:
i think the package has taken care of it, the addresses can only be: ipv4, ipv4 compatible ipv6 (deprecated) or ipv6.  the first two cases will return non-nil.   then we also check whether the address is a loopback address, ipv6 loopback need to exactly match `IPv6loopback               = IP{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}`